org.eclipse.scanning.example/src/org/eclipse/scanning/example/malcolm/DummyMalcolmRecord.java

```diff
@@ -0,0 +1,453 @@
+package org.eclipse.scanning.example.malcolm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.epics.pvaccess.server.rpc.RPCResponseCallback;
+import org.epics.pvaccess.server.rpc.RPCServiceAsync;
+import org.epics.pvaccess.server.rpc.Service;
+import org.epics.pvdata.factory.FieldFactory;
+import org.epics.pvdata.factory.PVDataFactory;
+import org.epics.pvdata.factory.StatusFactory;
+import org.epics.pvdata.pv.FieldBuilder;
+import org.epics.pvdata.pv.FieldCreate;
+import org.epics.pvdata.pv.PVBoolean;
+import org.epics.pvdata.pv.PVBooleanArray;
+import org.epics.pvdata.pv.PVDataCreate;
+import org.epics.pvdata.pv.PVDoubleArray;
+import org.epics.pvdata.pv.PVFloat;
+import org.epics.pvdata.pv.PVFloatArray;
+import org.epics.pvdata.pv.PVInt;
+import org.epics.pvdata.pv.PVIntArray;
+import org.epics.pvdata.pv.PVString;
+import org.epics.pvdata.pv.PVStringArray;
+import org.epics.pvdata.pv.PVStructure;
+import org.epics.pvdata.pv.PVUnion;
+import org.epics.pvdata.pv.ScalarType;
+import org.epics.pvdata.pv.Status;
+import org.epics.pvdata.pv.Status.StatusType;
+import org.epics.pvdata.pv.Structure;
+import org.epics.pvdata.pv.Union;
+import org.epics.pvdatabase.PVDatabase;
+import org.epics.pvdatabase.PVDatabaseFactory;
+import org.epics.pvdatabase.PVRecord;
+
+class DummyMalcolmRecord extends PVRecord {
+	
+    private static final FieldCreate fieldCreate = FieldFactory.getFieldCreate();
+    private static final PVDataCreate pvDataCreate = PVDataFactory.getPVDataCreate();
+
+    private boolean     underControl = false;
+    
+    private Map<String, PVStructure> receivedRPCCalls = new HashMap<String, PVStructure>();

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)